src/write.c

```diff
@@ -941,6 +941,30 @@ static avifCodecType avifEncoderGetCodecType(const avifEncoder * encoder)
     return avifCodecTypeFromChoice(encoder->codecChoice, AVIF_CODEC_FLAG_CAN_ENCODE);
 }
 
+static avifBool avifEncoderDataForceKeyframeForAlpha(const avifEncoderData * data, const avifEncoderItem * colorItem, avifAddImageFlags addImageFlags)
+{
+    if (!data->alphaPresent) {
+        // There is no alpha plane.
+        return AVIF_FALSE;
+    }
+    if (addImageFlags & AVIF_ADD_IMAGE_FLAG_SINGLE) {
+        // Not an animated image.
+        return AVIF_FALSE;
+    }
+    if (data->frames.count == 0) {
+        // This is the first frame, no need to force keyframe.
+        return AVIF_FALSE;
+    }
+    const uint32_t sampleCount = colorItem->encodeOutput->samples.count;
```

Comment:
Nit: It would also be helpful to document that colorItem->encodeOutput->samples stores the encoded frames of the color channel that the encoder has output so far.

Reply:
I changed the variable name to `colorFramesOutputSoFar`.